### PR TITLE
feat: support XML vs. JSON in throughput_vs_cpu_benchmark

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/4277)

### CODE REVIEW

File: google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc [hunk 1]
```diff
@@ -268,6 +285,9 @@ TestResults RunThread(Options const& options, std::string const& bucket_name) {
   std::bernoulli_distribution crc_generator;
   std::bernoulli_distribution md5_generator;
 
+  std::uniform_int_distribution<std::size_t> api_generator(
+      0, options.enabled_apis.size() - 1);
```
Comment (devbww) [inline google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc:289]: Should we have checked `!options.enabled_apis.empty()` somewhere?

File: google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc [hunk 2]
```diff
@@ -412,51 +464,61 @@ google::cloud::StatusOr<Options> ParseArgsDefault(
...
-                                 std::move(os).str()};
+    return make_status(os);
   }
 
   if (!gcs_bm::SimpleTimer::SupportPerThreadUsage() &&
       options.thread_count > 1) {
+    std::cerr <<
+        R"""(
+# WARNING
+# Your platform does not support per-thread usage metricsand you have enabled
```
Comment (devbww) [inline google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc:510]: s/metricsand/metrics and/

File: google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc [hunk 3]
```diff
@@ -412,51 +464,61 @@ google::cloud::StatusOr<Options> ParseArgsDefault(
...
+# multiple threads, the CPU usage results will not be usable. See getrusage(2)
```
Comment (devbww) [inline google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc:511]: s/the/so the/?

File: google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc:289]: That is done in `ParseArgsDefault()` along with all other argument validation.

File: google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc]: Fixed.

File: google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc (hunk 3 above)
Comment (coryan) [reply google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc]: Fixed.

File: google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc (hunk 1 above)
Comment (devbww) [reply google/cloud/storage/benchmarks/storage_throughput_vs_cpu_benchmark.cc:289]: Oops ... I see it now.  Thanks.

**Comment by coryan** [comment]
Argh, I forgot to say this is part of #4273

**Pull request merged**